tools/src/h5dump/h5dump.c

```diff
@@ -149,6 +149,29 @@ static struct h5_long_options l_opts[] = {{"attribute", require_arg, 'a'},
                                           {"vfd-info", require_arg, '6'},
                                           {NULL, 0, '\0'}};
 
+#if defined(H5_HAVE_PARALLEL) && defined(H5_HAVE_SUBFILING_VFD)
```

Comment:
Rather than needing to call MPI_Init in every serial tool that wants to support parallel VFDs/VOL connectors, I think it makes more sense to remove the check for whether MPI is initialized in the tools library and then document that a parallel VFD or VOL connector should initialize MPI itself if it needs it and it wasn't already initialized. However, this assumes that systems which were previously broken in this regard (Theta IIRC) are no longer problematic due to the tools trying to open a file with any available VFD when the first attempt fails. In fact, it may even be appropriate to remove that functionality from the tools at this point.

Regardless, we shouldn't generally have functionality in the tools with an #ifdef for a particular VFD.